Odata-docs/odatalib/v7/edm/iedmtargetpath-for-annotations-target.md

```diff
@@ -0,0 +1,463 @@
+
+            this.Model.AddElement(this.Container);
+        }
+
+        public EdmModel Model { get; private set; }
+        public EdmEntityContainer Container { get; private set; }
+        public EdmEntitySet EntitySet { get; private set; }
+        public EdmSingleton Singleton { get; private set; }
+        public EdmEntityType Customer { get; private set; }
+        public EdmEntityType VipCustomer { get; private set; }
+        public EdmEntityType City { get; private set; }
+        public EdmComplexType Address { get; private set; }
+        public EdmComplexType DerivedAddress { get; private set; }
+        public IEdmProperty NameProperty { get; private set; }
+        public IEdmProperty AddressProperty { get; private set; }
+        public EdmNavigationProperty NavUnderComplex { get; private set; }
+        public EdmNavigationProperty NavUnderCustomer { get; private set; }
+    }
+}
+
+```
+
+## Write out-of-line annotations
+
+Replace the contents of _Program.cs_ file with the following code:
+
+```csharp
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Vocabularies;
+using Microsoft.OData.Edm;
+using System.Reflection;
+using System.ComponentModel;
+using Microsoft.OData.Edm.Validation;
+using System.Xml;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+using System.Xml.Linq;
+
+namespace AnnotationsTarget
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            SampleModel sampleModel = new SampleModel();
+            EdmModel model = sampleModel.Model;
+
+            // Write Annotations
+            EdmTargetPath targetPath = new EdmTargetPath(sampleModel.Container, sampleModel.EntitySet, sampleModel.NameProperty);
+            EdmTerm term = new EdmTerm("NS", "MyTerm", EdmCoreModel.Instance.GetString(true));
+            model.AddElement(term);
+            EdmVocabularyAnnotation annotation = new EdmVocabularyAnnotation(targetPath, term, new EdmStringConstant("Name OutOfLine MyTerm Value"));
+            annotation.SetSerializationLocation(model, EdmVocabularyAnnotationSerializationLocation.OutOfLine);
+            model.SetVocabularyAnnotation(annotation);
+
+            WriteXml(model);
+        }
+
+        private static void WriteXml(IEdmModel model, CsdlTarget target = CsdlTarget.OData)
+        {
+            using (StringWriter sw = new StringWriter())
+            {
+                XmlWriterSettings settings = new XmlWriterSettings();
+                settings.Encoding = System.Text.Encoding.UTF8;
+                settings.Indent = true;
+
+                using (XmlWriter xw = XmlWriter.Create(sw, settings))
+                {
+                    IEnumerable<EdmError> errors;
+                    CsdlWriter.TryWriteCsdl(model, xw, target, out errors);
+                    xw.Flush();
+                }
+
+                string output = sw.ToString();
+                Console.Write(output);
+            }
+        }
+    }
+}
+```
```

Comment:
Before inviting the reader to run the application, you want to explain your code sample
- Describe what is happening in `InitializeModel` method. In brief since it's not the focus of your doc, e.g., In the `InitializeModel` method we're initializing the model... yada yada
- Describe what is happening in "AddOutOf..." method. In detail since that's the focus of your doc.
- Describe in 1 or 2 sentences what is happening in `SerializeModel` method.